server/boot/challenge.js

```diff
@@ -23,11 +23,15 @@ function buildUserUpdate(
 
 
   if (alreadyCompleted) {
+    if (!oldChallenge.hasOwnProperty('numOfAttempts')) {
+      oldChallenge.numOfAttempts = 1;
+    }
```

Comment:
This conditional does not protect against `undefined`. If this property is added to the model schema it will always be defined. 

A better solution is to define the `numOfAttempts` independently of the `oldChallenge` using a ternary expression.

```js
const numOfAttempts = oldChallenge.numOfAttempts ? oldChallenge.numberOfAttempts + 1 : 0;
```

Then use this value to as the value assigned to the finalChallenge declaration. This protects against `undefined` and ignores whether the previous challenge attempt had the property defined or not.